### ISSUE TITLE
GameDev na FIT

### ISSUE THREAD

**Issue by nightmaredev** [su-fit-vut/student-voice#20]
Jak už bylo řečeno, toto téma by si asi zasloužilo vlastní issue, tak je tady.

Uvědomuju si, že fakulta více méně učí většinu technických věcí, co je pro gamedev potřeba, avšak
mi přijde, že se ke hrám staví tak nějak moc neutrálně, až spíše negativně. Na FITu jsem neviděl o
GameDevArea ani letáček a na meetupech mě i dalším bylo řečeno, že FITu byly nabídnuty
přednášky nebo spolupráce, ale fakulta neměla zájem. 🙁Jsem si jist, že herně vývojářská
přednáška či série přednášek by přilákala daleko více lidí než přednášky IZP v našem první semestru nebo „vtipně“ nazvané přednášky typu “Using Pot and LSD to improve ACID transactions”.

Hry jsou nevyhnutelným krokem zábavního průmyslu a součást kultury. Jenom je škoda, že Česká
Republika vydá nějaký "wow!" titul jednou za 10let (jsem shovívavý a počítám mezi ně i mobilní
Dead Trigger) a furt mi přijde, že se na ně pohlíží jako „Root of all evil“, něco, co do seriózní, slušné
společnosti nebo konverzace nepatří, i přesto, že už dávno dominuje zábavní průmysl a každý si
doma zapne nějakou tu gamesu ať už mu je 10, 20 nebo 40.

Co bych navrhl ke změně:
1. Nestranit se tomu a uznat, že prostě hry do IT patří. Uspořádat, ale především povolit nějaké
přednášky na herní témata (ať už během konferencí nebo mimo). Nedávno v X-Zone *{0}
prodejně byl celkem dobře zpracovaný dokument o vývoji Kingdom Come. Tohle by pak
zvýšilo i atraktivitu fakulty v očích mladého studenta střední školy.

2. Uspořádat lokální Game Jam (herní hackathon) *{1} - studenti by se zlepšili v programování.
Přeci jenom je programování o zkušenosti a praxi, a každá zábavná forma kódování
studentovi pomůže si odpočinout od projektů a dodá mu chuť nebo motivaci, což je na FITu
občas velký problém. Student se často cítí, že se tu jenom snaží přežít a ne, že by se
vzdělával.

3. Jak už tu kolega psal u jednoho z postů, tak by bylo dobré více studentovi ukázat a dát
příklad, k čemu mu je představovaná látka a jak případně se prolínají mezi sebou. Nedávno
jsem řešil, jak si vykreslit data z prohledávací heatmapy pro AI v mé hře, ale v době, kdy jsme
dělali ISS projekt jsem si myslel, že mi to bude úplně k ničemu. Z projektu jsem si nakonec
vůbec nic nepamatoval a neodnesl a tak jsem to musel dělat v excelu. =( *{2}

Sám se rád zapojím do jakékoliv herní iniciativy. Můžete kouknout na něco z mé tvorby. Zároveň toto issue beru spíše jako diskuzní téma, tak snad se k tomu vyjádří více lidí, pokud by měli nějaké doplnění k tomu, co jsem zde uvedl. =) *{3}

* Odkazy:
_{0} – Deliverance: The Making of Kingdom Come a FB event
- https://store.steampowered.com/app/976180/Deliverance_The_Making_of_Kingdom_Come/?curator_clanid=32942828
- https://www.facebook.com/events/346678876060250/

{1} – Linky na Game jamy
- https://alakajam.com/
- https://globalgamejam.org/ (lokální gamejam, kde se předem musí domluvit lokace s pořadateli)
- https://ldjam.com/

{2} – heatmapa
https://i.imgur.com/s7ROpMO.gif

{3} – Osobní reklama
- www.icsharp.eu
- https://alakajam.com/user/josefgazdos
- https://globalgamejam.org/users/nightmare_

**Comment by StupydHors** [comment]
Musím souhlasit s názorem @nightmaredev.
To že se znalosti z předmětu někde dají využít nebo to že se naučíme tu znalost využít je zásadní rozdíl.
Praxe je nade vše a pokud by se fakulta stavěla k programování videoher vstřícněji myslím že by to zvýšilo její atraktivitu. Takový Game Jam by určitě mnoho lidí uvítalo (GlobalGameJam 2019 byl pořádán pouze v Praze což je škoda)
V ČR již pár programů na vývoj her existuje. Jak bylo zmíněno tak FI MUNI je v tomto napřed a od příštího roku bude poskytovat magisterské vzdělání zaměřující se na vývoj her. Mezitím mám dojem že FIT se tomuto trendu brání.

**Comment by Bordovsky** [comment]
Jsem za toto téma velmi rád, a budu vděčný každému kdo zde vyjádři nejaký názor k věci. Pokud budu mít podklady od studentů , že o takové věci opravdu stojí, pujde vyjednávání snáze. 

Spoustu věcí vidím podobně. Mým cílem pro tento semestr bylo kontaktovat GameDevArea a zkusit se dohodnout na nějakých hostovaných přednáškách. Má představa byla tři přednášky rozložené do celé délky semestru. Bohužel už je pulka semestru a krom diskuze s panem proděkanem pro vztahy s veřejností a jedné návštěvy GameDevMeetup jsem se k tomuto tématu nedostal. (O množství volného času které FIT poskytuje asi nemusím vyprávět.)
Přisto ale doufám, že se podaří něco ještě uspořádat. Pokud by fakulta viděla že je o takové přednášky zájem, a že jsou pro studenty přínosem (viz níže) určitě by byla nakloněna k tvorbě nějakého předmětu/semináře, například obdobně jako je předmět IAN (Analýza binárního kódu, vyučují lidé  z RedHatu) a časem se to může vyvinout dál.

Obavy fakulty jsou například v tom, že takové přednášky pořádané nějakou firmou mohou způsobit odliv studentů. Zlákat je k práci na atraktivních projektech (hrách), studenti začnou brigádou a místo tvorby projektů za body/kredity budou dělat hry za penize. 
Taktéž by se mělo jednat o přednášky s opravdovou přidanou hodnotou. Upřimě bastlit kód dnes zvládne kdejaky středoškolák co si přečetl tutorial o Javascriptu. Absolvent FIT by měl přeci jen umět něco vic, zhodnotit složitost použitých algoritmů, efektivitu využiti paměti a optimální přístupové vzory atd. 

Na GameDevMeetupu mě zarazilo množství otevřených pozic, které firmy mají a uvědomil jsem si, že některé firmy se opravdu mohou dívat na FIT jako na zdroj levné pracovní síly. Veřím ale, že se najdou i firmy které budou stavět předání znalostí na první místo a bude možné se s nima na něčem dohodnout. Lidé z průmyslu mohou nabídnout mnoho lákavých témat ať už z oblasti uměle inteligence, softwerového ingenýrství, počítačové grafiky a dalších.

Z mých informací FIT již v minulosti jednal s Bohemia Interactive o vzajemně vyhodné spolupraci ale jednání ztroskotala/vyšuměla do ztracena. Fakulta od svých partnerů vyžaduje podpis kodexu, který zavazuje firmu k poskytnutí studentského volna studentům a k motivaci k dokončení studia. (Manažeři některých firem s kterýma jsem se bavil se na nedokončené studium divají jako na znak neschopnosti dokončit projekt a značně limitují možnost postupu takových zaměstnanců. Pokud ale firma hledá jen "cvičene opice" co budou mlátit kód tak je jim nějake studium putna.) 

GameJam/Herni Hackaton je opět záležitost která se hodně špatně organizuje. Je potřeba mít dostatečné znalosti o tvorbě her, ideálně někdo z průmyslu. Pokud by fakulta uzavřela partnerskou smlouvu s nějakou firmou zabývající se takovou věcí, ideálně vícero firem, určitě by to bylo realizovatelné.

Ad HeatMap/OffTopic> V Excelu? Aby člověk, který absolvoval IZP musel šáhnout po takovémto řešení se mi zdá poněkud smutné...

**Comment by 0xFUCK** [comment]
Tiež som sa modernejšiu výuku game dev a celkovo aktuálnych vecí v IT. Princípy ponechajme, prinesme avšak niečo nové aj (dnes s kolesom ako ukážkou princípu nič moc neurobíme, svet požaduje viac - základné znalosti z aktuálneho riešenia danej problematiky sú nutnosť). Škoda že prednáška od človeka z praxe je skôr výnimkou na FITe.. 

Taktiež mi chýba priestor na výuku nových zaujímavých jazykov - Go, Rust. Pre sieťarov je Go dobrá voľba, v dnešnej dobe fakt nikto necanceruje  v C aby urobil bezpečnú server-klient aplikáciu.

**Comment by 0xFUCK** [comment]
@olokex dôvod palcu dole?

**Comment by 0xFUCK** [comment]
No ty si dobrý kokot, @olokex !

**Comment by nightmaredev** [comment]
Tak jsem byl včera na gamerpie. Skvělá akce jak organizačně, tak obsahově. Komukoliv koho zajímají alespoň trochu hry a dění okolo, tak doporučuju.  Přednášky byly na čas a hlavně byly všechny vcelku zajímavé. Co mě avšak zesmutnělo, že jediná známka VUT byl nějaký fotbálek s displayem a to ještě asi bylo ze strojárny. MUNI tam mělo obrovský advertisement. Myslím, že fakultě tímto neutíká pouze  řada studentů, ale i šílený randál šekelů, který by byli jak Bohemia Interactive, tak Madfinger games (ale i ostatní studia) do nějakých projektů/eventů určitě ochotni dát, jak je ostatně vidno právě na MUNI GameStudies i gamerpie... 
Abych ale jenom nenadával, položím otázku, s kým to na fakultě řešit?

**Comment by MarcusMorter** [comment]
@nightmaredev Mno, tak bych byl rád za nějaký ten GameDev, ono to kolikrát z boring věci, co se musíme nadrtit, dělá celkem zajímavou látku (např pravděbodobnost v matice, hned bych se o to zajímal víc, kdybych chtěl balancovat random dropy zbraní ve hře), a těch příkladů je hodně. Je to jedna z věcí, která fakultě opravdu uniká, už vzhledem k tomů, že gaming je zdaleka na prvním místě zábavního průmyslu a lidi se o něj aktivně zajímají. Zkusím se s ostatními z SU pobavit jak to vidí, možná by z toho něco vzešlo.

**Comment by nightmaredev** [comment]
Na ČTéčku vyšla reportáž o chybějících lidech v herním průmyslu.
https://www.ceskatelevize.cz/ivysilani/1097181328-udalosti/219411000100330/obsah/685045-pc-hry-tahounem-digitalni-ekonomiky?fbclid=IwAR1O8-nmDu40CkyKcYoTXhv-_7S9WpEAz-kApqoAzisV6EFEW5gB-Z_zZ1s

**Comment by Bordovsky** [comment]
> �Na ČTéčku vyšla reportáž o chybějících lidech v herním průmyslu.

Chybějící lidé v herním průmyslu jsou jako argument dvojsečná zbraň. Na jedné straně z dlouhodobého hlediska-> Chybí lidi v průmyslu, o absolventy takového zaměření bude zájem. Z krátkodobého hlediska ti lidi chybí teď, studenti půjdou pracovat a na studium budou kašlat.

@nightmaredev Jak jsem říkal. Záležitost (partneři fakulty, spolupráce s průmyslem) spadá pod pana proděkana pro vnější vztahy od nějž jsem na začátku zimy dostal stanovisko, že hernímu vývoji nestojí v cestě. Partnera z dané oblasti škola zatím nemá i když už jednala s Bohemia Interactive kde jednání vyšuměla. Pokud SU někoho osloví, jak bylo v mým plánem, může uspořádat přednášku, pokud o ni bude zájem a fakulta uvidí že je přenosná dá se na tom stavět dále. Nicméně tato přednáška by neměla být náborového charakteru, a měla by prokazatelně přinést přidanou hodnotu studentům (s čímž souhlasím). Další lidé s kterými by se tato záležitost měla řešit by byli pravděpodobně proděkani pro bakalářksé/magisterské studium. Z mého pohledu je nešťastné vytvořit obor na základě požadavků jedné firmy. Chtělo by to více partnerů z průmyslu, vytvořit plénum kde lidé řeknou co v průmyslu chybí, jaké jsou požadavky na absolventy a podobně. 

Bohužel mé současné povinnosti mi nedovolují věci se více věnovat. Pro zajímavost Asociace českých herních vývojářů vydala nedávno statistiku za rok 2017. Máme 76 herních vývojářů(studii), celkem s obratem 2.26 miliardy korun. (Dvojnásobek oproti filmu a televizi dohromady.) Za rok 2018 se čeká číslo nad 3 miliardy, dáky uspěchu KC:D a Beatsaber (to padlo už možna i v té reportáži v ČT, ale statistiky jsem ještě osobně neviděl).

Studijní obory jsou jen 3 v cele ČR zaměřeny přímo na vývoj počítačových her. Vím, že studenti kteří mají zájem pouze o tyto věci raději studijí v zahraničí kde tyto obory/programy už mají nějakou tradici. Podpora ze strany českého státu je momentálně nízká až nulová (alespoň co je mi známo). Tuším, že pan ministr Bělobrádek za KDU-ČSL měl nějaké meetingy ohledně game-dev a podpory od státu někdy kolem 2017, jenže od té doby se v politice dějí neustále nějaké věci a myslím, že ať plánoval/sliboval cokoli, je to v současnosti u ledu.

Situace se bude pravděpodobně zlepšovat, tento segment IT průmyslu již zdaleka není zanedbatelný. Otázkou je, kdy si to uvědomí FIT a začne se tomu věnovat. Město Brno tuším chystá nějakou podporu pro začínající herní vývojáře (mluvilo se o nějakých levných nájmech pracovních prostor) zároveň s podporou od velkých firem (můžou poskytnout know-how). Na FIT existuje startup@FIT (ale nevím, jestli by gamedev spadal do podmínek které tam mají, podrobnosti o tom jsem nezjišťoval).

Během mé diskuze s panem proděkanem padlo, že FIT nechce prostě jen k něčemu plácnout nálepku/buzzword "game-dev", ale že to co se tady učí tam spadá. (To, že se v současnosti plánuje studijní program s buzzword umělá inteligence nechám bez komentáře.)

Ohledně "aplikovatelnosti" znalostí z předmětů na FIT....(povzdech) to nevidím jen jako problém GameDev.
Přednášející by měl sám motivovat k tomu aby se učili danou látku a o její užitečnosti by měl studenty přesvědčit. Někteří tak činí, jiní ne (bohužel asi většina). Já měl to štěstí, že jsem studoval Vestavěné a počítačové systémy, kde většina předmětů svůj impakt jasně demonstrovala. Někdy, zvlášť u základů jako je matematika, pravděpodobnost a podobně, je problém ten, že uplatnit lze všude. Kdyby vyučující vybral jen jednu z nich, například GameDev, lidé kteří se zajímají o jiné věci, analýzu trafiku, kódování, šifrování, detekci vzorů a miliony dalších, budou ochuzeni a GameDev je nezaujme.

**Comment by Bordovsky** [comment]
> @nightmaredev Mno, tak bych byl rád za nějaký ten GameDev, ono to kolikrát z boring věci, co se musíme nadrtit, dělá celkem zajímavou látku (např pravděbodobnost v matice, hned bych se o to zajímal víc, kdybych chtěl balancovat random dropy zbraní ve hře), a těch příkladů je hodně. Je to jedna z věcí, která fakultě opravdu uniká, už vzhledem k tomů, že gaming je zdaleka na prvním místě zábavního průmyslu a lidi se o něj aktivně zajímají. Zkusím se s ostatními z SU pobavit jak to vidí, možná by z toho něco vzešlo.

Věc v rámci SU řeším já, tohle téma vytahuji už přes rok . Určitě jsi toho všiml pokud chodíš na zasedání SU a dáváš pozor. 
Určitě máš pravdu že herní sektor je výraznou, ne-li nejvyděleční části zábavního sektoru, nicméně si stále myslím že podíl IT na industriálním sektoru to překonává a byla by chyba cely FIT orientovat jen na hry. Navíc na každou úspěšnou hru máš desítky nespěšných projektů které se na trhu nechytly... Sázka na vývoj her je dost loterie. Může se to vrátit jako u BeatSaber a nebo tě může hejtit komunita tak jako u No Man's Sky a budeš se hrabat v problémech.

**Comment by nightmaredev** [comment]
No, jistě. Já neříkám, že je hned třeba založit obor nebo volitelný předmět tomu věnující se (ikdyž by volitelný předmět možná nebyl na škodu). Určitě mi ani nejde o to překopávat FIT na herní fakultu. Spíše se pokusit uspořádat přednášku, případně workshop. Pokud tedy říkáš, že proděkan hernímu vývoji nestojí v cestě, můžu se pokusit otázat některých lidí v průmyslu, jestli by neměli zájem něco přednést.

**Comment by davidjaho** [comment]
Pokud mohu přidat svou trošku do mlýna, tak přednášky hostů z praxe si dokáži dost dobře představit. Kdyby byly například ve stylu jako na GDS*{1}{2}, myslím si, že by zaujali nemalou část studentů. Avšak musely by se překonat výše popsaná úskalí to bez debaty.

Volitelný seminář zabývající se problematikou vývoje her by nejspíš také nebyl úplně nerealizovatelný. Například kdyby se přistoupilo k formátu týmového projektu IVS. Tam by se přeci jen dali skloubit znalosti z mnoha předmětů (např. umělá inteligence, grafika, databáze, atd.) Samozřejmě by se to muselo hodně promyslet, dát tomu vlastní formu a tak, ale možnosti tu jistě jsou.

PS: Kde jsou dnes ti co jednu dobu diskutovali o vytvoření fiťáckých Divokých kmenů? :-) 

{1} GDS - http://www.gdsession.com/
{2} GDS záznamy - https://www.youtube.com/user/GamesczGDS2012/videos

**Comment by nightmaredev** [comment]
No. Abychom mohli udělat tak kvalitní přednášky jako na GDS, tak bychom museli mít přímo takové legendy jako na GDS. Jen málokdo zvládne udělat tak zábavou přednášku jako třeba Viktor Bocan. Osobně bych rád přispěl do mlýnku, avšak si nejsem jist, čím bych směl.

**Comment by Bordovsky** [comment]
> ..... avšak si nejsem jist, čím bych směl.

Existuji jisté omezení, ale pokud si promluvíš s lidmi kteří za nimi stojí a jsi ochoten se na věc podívat z jejich pohledu tak brzo zjistíš, že mají docela dobrý důvod. 

Netýká se to jen GameDev...  

Podpořit takový nápad je jedna věc, dá se pak prezentovat fakultě, že je o takovou věc zájem. (I když v současné době bych mohl akorát říct že o GameDev má krom členů SU zájem jen nějací 3 lidi.)
Nicméně (a trochu se opakuji) kde nejsou lidi kteří jsou ochotni se prohrabat tím, co se smí a nesmí, a zařizovat takové věci, nic jiného než mít to na poličce dobrých nápadů nezmůžeme.
